RecoHI/HiTracking/plugins/HIPixelTracksSlimmer_miniAOD.cc

```diff
@@ -0,0 +1,256 @@
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/TrackReco/interface/TrackFwd.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+
+#include "DataFormats/Candidate/interface/Candidate.h"
+#include "DataFormats/RecoCandidate/interface/RecoChargedCandidate.h"
+#include "DataFormats/PatCandidates/interface/PackedCandidate.h"
+#include "DataFormats/Common/interface/Association.h"
+
+#include <vector>
+
+//
+// class declaration
+//
+
+class HIPixelTracksSlimmer_miniAOD : public edm::stream::EDProducer<> {
+public:
+  explicit HIPixelTracksSlimmer_miniAOD(const edm::ParameterSet&);
+  ~HIPixelTracksSlimmer_miniAOD() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+  void addPackedCandidate(std::vector<pat::PackedCandidate>& cands,
+                          const reco::TrackRef& trk,
+                          const reco::VertexRef& pvSlimmed,
+                          const reco::VertexRefProd& pvSlimmedColl,
+                          bool passPixelTrackSel) const;
+
+private:
+  void beginStream(edm::StreamID) override;
+  void produce(edm::Event&, const edm::EventSetup&) override;
+  void endStream() override;
+
+  //virtual void beginRun(edm::Run const&, edm::EventSetup const&) override;
+  //virtual void endRun(edm::Run const&, edm::EventSetup const&) override;
+  //virtual void beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
+  //virtual void endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
+
+  // ----------member data ---------------------------
+  const edm::EDGetTokenT<reco::TrackCollection> srcTracks_;
+  const edm::EDGetTokenT<reco::VertexCollection> srcVertices_;
+  const edm::EDGetTokenT<reco::VertexCollection> srcPrimaryVertices_;
+  const double dzSigCut_;
+  const double dxySigCut_;
+  const double dzSigHP_;
+  const double dxySigHP_;
+  const double ptMax_;
+  const int covarianceVersion_;
+  const int covarianceSchema_;
+};
+
+//
+// constants, enums and typedefs
+//
+
+//
+// static data member definitions
+//
+
+//
+// constructors and destructor
+//
+HIPixelTracksSlimmer_miniAOD::HIPixelTracksSlimmer_miniAOD(const edm::ParameterSet& iConfig)
+    : srcTracks_(consumes<reco::TrackCollection>(iConfig.getParameter<edm::InputTag>("srcTracks"))),
+      srcVertices_(consumes<reco::VertexCollection>(iConfig.getParameter<edm::InputTag>("srcVertices"))),
+      srcPrimaryVertices_(consumes<reco::VertexCollection>(iConfig.getParameter<edm::InputTag>("srcPrimaryVertices"))),
+      dzSigCut_(iConfig.getParameter<double>("dzSigCut")),
+      dxySigCut_(iConfig.getParameter<double>("dxySigCut")),
+      dzSigHP_(iConfig.getParameter<double>("dzSigHP")),
+      dxySigHP_(iConfig.getParameter<double>("dxySigHP")),
+      ptMax_(iConfig.getParameter<double>("ptMax")),
```

Comment:
perhaps add ptMin, so that this module can be more useful directly for possible other purposes

Reply:
I did this and also I included a parameter "resetHP" in order to be more general for other track collections that have HP flag set. This comment is included in this commit: https://github.com/cms-sw/cmssw/pull/30313/commits/deebf6766c730a8c97ac3f3fe90b065b24d5e70d